sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java

```diff
@@ -28,27 +39,128 @@
+                .invokeWithArguments();
+        } catch (Exception | LinkageError e) {
+            LOGGER.log(LogLevel.VERBOSE, () -> "Unable to reflective locate or access "
+                + "'com.azure.core.experimental.implementation.rest.SyncRestProxyExperimental'. Enablement of "
+                + "'AZURE_HTTP_REST_PROXY_EXPERIMENTAL_ENABLED' or 'com.azure.core.http.restproxy.experimental.enable' "
+                + "will be ignored and the base 'SyncRestProxy' will be used instead.", e);
+        } catch (Throwable e) {
+            // At this point the only type left should be Error, rethrow it as an Error.
+            throw (Error) e;
+        }
+        SYNC_EXPERIMENTAL_CREATOR = syncExperimentalCreator;
+
+        ParserExperimentalConstructor parserExperimentalConstructor = null;
+        try {
+            Class<?> swaggerExperimentalClass = Class.forName(
+                "com.azure.core.experimental.implementation.rest.SwaggerInterfaceParserExperimental");
+            MethodHandles.Lookup lookup = ReflectionUtils.getLookupToUse(swaggerExperimentalClass);
+            MethodHandle unreflectedFactory = lookup.unreflect(swaggerExperimentalClass.getDeclaredMethod(
+                "getInstance", Class.class));
+            parserExperimentalConstructor = (ParserExperimentalConstructor) LambdaMetafactory.metafactory(
+                    MethodHandles.lookup(), "apply", ParserExperimentalConstructor.METHOD_TYPE,
+                    ParserExperimentalConstructor.SIGNATURE, unreflectedFactory,
+                    ParserExperimentalConstructor.SIGNATURE)
+                .getTarget()
+                .invokeWithArguments();
+        } catch (Exception | LinkageError e) {
+            LOGGER.log(LogLevel.VERBOSE, () -> "Unable to reflective locate or access "
+                + "'com.azure.core.experimental.implementation.rest.SwaggerInterfaceParser'. Enablement of "
+                + "'AZURE_HTTP_REST_PROXY_EXPERIMENTAL_ENABLED' or 'com.azure.core.http.restproxy.experimental.enable' "
+                + "will be ignored and the base 'SwaggerInterfaceParser' will be used instead.", e);
+        } catch (Throwable e) {
+            // At this point the only type left should be Error, rethrow it as an Error.
+            throw (Error) e;
+        }
+        PARSER_EXPERIMENTAL_CONSTRUCTOR = parserExperimentalConstructor;
+        CAN_USE_EXPERIMENTAL = ASYNC_EXPERIMENTAL_CREATOR != null && SYNC_EXPERIMENTAL_CREATOR != null
+            && PARSER_EXPERIMENTAL_CONSTRUCTOR != null;
+    }
+
+    private final Class<?> swaggerInterface;
     private final SwaggerInterfaceParser interfaceParser;
+    private final SwaggerInterfaceParser interfaceParserExperimental;
     private final AsyncRestProxy asyncRestProxy;
     private final HttpPipeline httpPipeline;
     private final SyncRestProxy syncRestProxy;
+    private final AsyncRestProxy asyncRestProxyExperimental;
+    private final SyncRestProxy syncRestProxyExperimental;
```

Comment:
Suffix is there so RestProxy can manage both an experimental and regular version of both async and sync proxies. But we could change this so there is just a global switch instead of both a global and instance switch and merge the four proxies into two

Reply:
I'd not duplicate fields to achieve that. They're of same types (experimental or not). After RestProxy construction it fields should have base or experimental implementation and RestProxy should be agnostic to that (except ctor/factory that creates it). If we still need to differentiate them in the RestProxy logic then inheritnace is not the best tool here.